lab_1/main.py

```diff
@@ -45,7 +44,19 @@ def calculate_frequencies(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    frequencies = {}
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+        if token not in frequencies.keys():
```

Comment:
is there a shorter way to check it?

Reply:
I've put `.isinstance()` check in the beginning of the function and combined it with `for` loop using `all`

Reply:
no, I mean exactly this line. It can be shorter